staking/README.md

```diff
@@ -0,0 +1,541 @@
+
+## Table of Contents
+
+* [Preliminary terminology](#preliminary-terminology)
+* [Overview](#overview)
+* [How does one join the network as a validator](#how-does-one-join-the-network-as-a-validator)
+* [How to propose a MsgValidatorJoin transaction](#how-to-propose-a-msgvalidatorjoin-transaction)
+* [How does an existing validator exit the network](#how-does-an-existing-validator-exit-the-network)
+* [How to propose a MsgValidatorExit transaction](#how-to-propose-a-msgvalidatorexit-transaction)
+* [How does a validator update its stake](#how-does-a-validator-update-its-stake)
+* [How to propose a MsgStakeUpdate transaction](#how-to-propose-a-msgstakeupdate-transaction)
+* [How does a validator update its signer address](#how-does-a-validator-update-its-signer-address)
+* [How to propose a MsgSignerUpdate transaction](#how-to-propose-a-msgsignerupdate-transaction)
+* [Query commands](#query-commands)
+
+## Preliminary terminology
+
+* An `epoch` represents the period till a checkpoint is submitted on Ethereum, i.e one `epoch` ends when a checkpoint is committed on Ethereum and the next one begins.
+
+## Overview
+
+The `staking` module in Heimdall is responsible for a validator's stake related operations. It primarily aids in
+
+* A node joining the protocol as a validator.
+* An node leaving the protocol as a validator.
+* Updating an existing validator's stake in the network.
+* Updating the signer address of an existing validator.
+
+## How does one join the network as a validator
+
+The node that wants to be a validator stakes its tokens by invoking the `stakeFor` method on the `StakeManager` contract on L1(Ethereum), which emits a `Staked` event:
+
+```
+/// @param signer validator address.
+/// @param validatorId unique integer to identify a validator.
+/// @param nonce to synchronize the events in heimdall.
+/// @param activationEpoch validator's first epoch as proposer.
+/// @param amount staking amount.
+/// @param total total staking amount.
+/// @param signerPubkey public key of the validator
+event Staked(
+    address indexed signer,
+    uint256 indexed validatorId,
+    uint256 nonce,
+    uint256 indexed activationEpoch,
+    uint256 amount,
+    uint256 total,
+    bytes signerPubkey
+);
+```
+
+An existing validator on Heimdall catches this event and sends a `MsgValidatorJoin` transaction, which is represented by the data structure:
+
+```
+type MsgValidatorJoin struct {
+	From            hmTypes.HeimdallAddress `json:"from"`
+	ID              hmTypes.ValidatorID     `json:"id"`
+	ActivationEpoch uint64                  `json:"activationEpoch"`
+	Amount          sdk.Int                 `json:"amount"`
+	SignerPubKey    hmTypes.PubKey          `json:"pub_key"`
+	TxHash          hmTypes.HeimdallHash    `json:"tx_hash"`
+	LogIndex        uint64                  `json:"log_index"`
+	BlockNumber     uint64                  `json:"block_number"`
+	Nonce           uint64                  `json:"nonce"`
+}
+```
+where ,
```

Comment:
I'd remove the comma

Reply:
So did I: [159a33b](https://github.com/maticnetwork/heimdall/pull/1082/commits/159a33bafa2f43c7cc2ae0f9d57b4257ef036436)